starport_template/lib/pages/sign_transaction_page.dart

```diff
@@ -124,10 +125,14 @@ class _SignTransactionPageState extends State<SignTransactionPage> {
   void _showAssetsTransferSheet(BuildContext context) {
     showMaterialModalBottomSheet(
       context: context,
+      isDismissible: false, // Stop user from tapping outside sheet to dismiss
       backgroundColor: Colors.transparent,
       builder: (context) => SizedBox(
         height: MediaQuery.of(context).size.height / 2.24,
         child: AssetsTransferSheet(
+          balance: widget.balance.copyWith(
+            amount: Amount.fromDouble(recipientGetsAmount),
+          ),
```

Comment:
let's not manipulate data inside the widget's tree. You can create a getter for the `reicipientGetsAmount` balance:

```dart
Balance get recipientGetsAmountBalance => widget.balance.copyWith(
            amount: Amount.fromDouble(recipientGetsAmount),
          );
```

Reply:
Here we are using the `fromDouble` constructor for `Amount` which we shouldn't. The `recipientGetsAmount` is a decimal value coming from this line somewhere above this file
`double get recipientGetsAmount => widget.transaction.amount.value.toDouble() - widget.transaction.fee;`

Technically this manipulation shouldn't be inside the UI and secondly this `widget.transaction.fee` should be a type of `Amount` as well so that we can add/subtract within decimals. What do you think @andrzejchm ?

Reply:
good catch @wal33d006 ! definetely we should stick with `Amount/Decimal` and not convert back and forth from/to doubles